### PR TITLE
Documentation reorg / edit suggestions + recommend `use_pkgdown_github_pages()` over `use_pkgdown()`

### PR DESCRIPTION
**Pull request by olivroy**
Addresses partly #2560.

The goal is to have a consistent way to access how each component of `_pkgdown.yml` can be customized.

* ~Join `init_site()` and `build_favicons()` docs. For consistency, `build_favicons()` should probably be `init_favicons()`?~
* Add a couple of references throughout. (some things are now very well documented in https://r-pkgs.org, so it'd be worth adding couple references to that!)

I tried gathering as much info as possible of potential allowed yaml config and create a menu item. (I didn't commit this, but here is what I have so far). (It would be great to have yaml validation in RStudio like we have with `_quarto.yml` for `_pkgdown.yml`, but that's probably a long shot.

From my findings, the info is all present, but is scattered between customize.Rmd, pkgdown.Rmd, `build_site()` and the different `build_*()` functions.

````yaml
navbar:
  structure:
    left: [intro, reference, articles, parts, news]
  components:
    parts:
      text: "Components of _pkgdown.yml"
      menu:
      - text: "Overview"
        href: reference/build_site.html#general-config
      - text: "-----"
      - text: "reference: (Package index)"
        href: reference/build_reference.html#reference-index
      - text: "figures: (How figures are rendered)"
        href: reference/build_reference.html#figures
      - text: "navbar: (Menus + components)"
        href: articles/customise.html#navbar-heading
      - text: "home (Homepage customization)"
        href: reference/build_home.html
      - text: "template: (Site theme and customisation)"
        href: '`vignette("customise")`'
      - text: "redirects: (Link redirects)"
        href: reference/build_redirects.html
      - text: "development: (Development mode)"
        href: reference/build_site.html#development-mode
      - text: "search: (Search functions)"
        href: reference/build_search.html
      - text: "news: (Changelog + release blog post)"
        href: reference/build_news.html
      - text: "lang: (Change the language)"
        href: articles/translations.html
      - text: "footer: (Personalize footer)"
        href: articles/customise.html#footer
      - text: "url:, destination:, title: (General Config)"
        href: reference/build_home.html#general-config
      - text: "repo: (Source repository)"
        href: reference/build_site.html#source-repository
      - text: "---"
      - text: "Examples"
        href: index.html#in-the-wild
      - text: "---"
      - text: "Template for articles or reference index examples"
        href: reference/templates.html
````

Preview:
image

### CODE REVIEW

File: R/build-favicons.R
Comment (hadley) [inline R/build-favicons.R:3]: I agree that `build_favicons()` is probably better as `init_favicons()` since it does imply that you don't need to run it very often. But I think you're likely to run it even less than `init_site()` (and indeed in most cases pkgdown will run it for you) so I think documenting with `init_site()` makes it maybe a bit too prominent.

File: R/build-home.R
Comment (hadley) [inline R/build-home.R:21]: Can you figure out what setting is trimming these trailing whitespaces and either turn it off or apply it to every file in one PR? Otherwise it makes it a bit hard for me to quickly scan and find the main differences.

File: R/build-search-docs.R
Comment (hadley) [inline R/build-search-docs.R:91]: There's little enough in that vignette, I think I'd almost prefer going in the opposite direction — just point to `?build_search` from the vignette. (And then maybe remove the vignette from the main navbar since it will only apply to BS3. Or maybe even remove it altogether with a redirect since BS3 is deprecated?)

File: pkgdown/_pkgdown.yml
Comment (hadley) [inline pkgdown/_pkgdown.yml:66]: I do think `init_site()` belongs above since the css that it builds is a site "component"

File: vignettes/pkgdown.Rmd
Comment (hadley) [inline vignettes/pkgdown.Rmd:176]: I'm not sure if it's worth recommending this at all any more.

File: pkgdown/_pkgdown.yml
Comment (hadley) [inline pkgdown/_pkgdown.yml:56]:
```suggestion
    for an overview.
```

File: vignettes/pkgdown.Rmd
Comment (hadley) [inline vignettes/pkgdown.Rmd:142]:
```suggestion
If you want to include an article on the website but not in the package (e.g., because it's large), you can either place it in a subdirectory of `vignettes/` (e.g. `vignettes/web_only`) or add it to `.Rbuildignore` (and make sure that there's no `vignettes:` section in the yaml header). See  `usethis::use_article()` for an automated way to do this.
```

**Review by hadley — APPROVED** [review]
Thanks for working on this! A bunch of smaller comments below.

File: R/build-home.R
Comment (olivroy) [reply R/build-home.R:21]: It is in .Rproj setting, when I save a file, it trims spaces... I don't know why this didn't happen for other contributors.

image

Possibly relates to https://github.com/rstudio/rstudio/issues/9902

File: R/build-search-docs.R
Comment (olivroy) [reply R/build-search-docs.R:91]: Sounds good, I will revert this change and I can open a new issue so that you can make the final decision.

File: R/build-home.R
Comment (olivroy) [reply R/build-home.R:21]: I'd recommend merging #2566 first, so that I can merge and I can take a last look at this PR before merge.

Comment (hadley) [reply R/build-home.R]: Oh yeah, it's probable that I have this turned off because it tends to cause problems when I drop into some random repo to make a tactical fix.

Comment (olivroy) [reply R/build-home.R]: I will revisit later with fresh eyes this PR with your comments in mind and will let you know when it is ready for final review!

**Comment by olivroy** [comment]
Not as thorough as I initially envisioned it, but some other improvements were made in the past 2 weeks. I think these are still improvements, so I think this PR can go as is. I can always send more later!

**Review by hadley — APPROVED** [review]
Feel free to merge whenever you're ready 😄

**Comment by olivroy** [comment]
@hadley I ended up pushing extra changes . f3cf62b and 8f3a286

* Typo fix `has_keywords()` -> `has_keyword()`, `pr_submit()` -> `pr_push()`
* Avoid mentioning Rbuildignore in the pkgdown documentation
* Rephrase the favicons topic and move it down to helpers in function organization.
* Add `@family` to `build_home()`, `build_articles()`, `build_news()`, and `build_reference()` for 2 reasons
  1. Add cross-refs between these topics. (I tried to rename topics to `home`, `articles`, `news`, and `reference`, but ran into a roxygen2 issue. (I will try providing a reprex there)
  2. Be able to use `has_concept()` in `_pkgdown.yml`. I think it is an underused / underadvertised feature, because it allows organizing your docs in R itself + pkgdown (instead of having to do it twice) What better way to learn about pkgdown than to look at pkgdown's `_pkgdown.yml` itself!

I will request a new review since these changes are substantial (and may be a question of preference)

File: README.Rmd
Comment (hadley) [inline README.Rmd:72]:
```suggestion
At last count, pkgdown is used [by over 12,000 packages](https://github.com/search?q=path%3A_pkgdown.yml+language%3AYAML&type=code&l=YAML). Here are a few examples:
```

File: pkgdown/_pkgdown.yml
Comment (hadley) [inline pkgdown/_pkgdown.yml:66]: The only downside here is that we lose the alphabetical ordering, but I don't think that's a big deal.

File: pkgdown/_pkgdown.yml
Comment (hadley) [inline pkgdown/_pkgdown.yml:100]: You can't redirect only specific fragments.

File: vignettes/pkgdown.Rmd
Comment (hadley) [inline vignettes/pkgdown.Rmd:121]:
```suggestion
If you want to include an article on the website but not in the package (e.g., because it's large), you can either place it in a subdirectory of `vignettes/` (e.g. `vignettes/articles`) or add it to `.Rbuildignore` (and make sure that there's no `vignettes:` section in the yaml header). See `usethis::use_article()` for an automated way to do this.
```

Do we really need to suggest two ways to do this?

File: vignettes/pkgdown.Rmd
Comment (hadley) [inline vignettes/pkgdown.Rmd:150]: Since we're now only recommending one way (which is good!), this sentence needs to be tweaked, and I think we can get rid of the bullets.

**Review by hadley — APPROVED** [review]

File: vignettes/pkgdown.Rmd
Comment (olivroy) [reply vignettes/pkgdown.Rmd:150]: I also made changes in README to reflect that everywhere!

File: README.md
Comment (olivroy) [inline README.md:78]:
```suggestion
At last count, pkgdown is used [by over 12,000
```

**Comment by olivroy** [comment]
@hadley do you want to take one final look as I made tweaks to vignette and README in c2a6389 to make the `use_pkgdown_github_pages()` advice more prominent everywhere?

**Review by hadley — APPROVED** [review]

**Pull request merged**
